### ISSUE TITLE
Getting started: YOLOX inference and image preprocessing

### ISSUE THREAD

**Issue by shayamram** [Deci-AI/super-gradients#778]
cc: @shaydeci 
I want to start using YOLOX for Object Detection using the pre-trained coco weights.
Looking at the [tutorial page](https://github.com/Deci-AI/super-gradients/blob/96d9c093e795b49149c69caf10cbfb7a4f1df955/tutorials/SG_transfer_learning_object_detection.ipynb) I was unclear if the expected input to the model should be **RGB** or **BGR**.

I looked at `COCODetectionDataset(DetectionDataset)` which lead me to the `.get_resized_image()` method. This method uses `cv2.imread`, which is BGR. So, is it correct to assume that the YOLOX pretrained model also had this kind of preprocessing? (i.e., the same as in super_gradients.training.datasets.detection_datasets.detection_dataset.DetectionDataset.get_resized_image)? 
While `COCODetectionDataset` seems to be using cv2 (-->BGR), the example in the tutorial above is using PIL.Image.open() which returns RGB.

Currently, I'm using `super_gradients.training.transforms.transforms.rescale_and_pad_to_size` to preprocess my images, but they are read by opencv (BGR), so I'm wondering if I need to skip the .swap() phase?

Thanks

**Comment by shaydeci** [comment]
Hey @shayamram (:

I assume you are pointing to the "Predict" cell, notice the line:
```
image_tensor = torch.tensor(np.array(image)[:, :, ::-1].copy()).permute(2, 0, 1).unsqueeze(dim=0).float().cuda()
```
The [:, :, ::-1] part reverses the order of the 3rd dimension, and therefore converts it to BGR format.
I will add a comment to clarify that this is also being done there.

If you have any other questions, please let me know.

**Comment by shayamram** [comment]
So to clarify: the model predict expects BGR *and* padded image? (example: the [image was padded](https://deci-pretrained-models.s3.amazonaws.com/sample_images/sample_image_person_car.jpg), I assume to maintain aspect ratio).

This seems different from the [example here](https://github.com/Deci-AI/super-gradients/blob/master/tutorials/SuperGradients%20How%20To%20Use%20a%20model%20to%20Predict.ipynb) (the one with the dog). This example is RGB and only simple resize (no padding), so I just want to be extra-certain that the correct preprocessing for prediction is (resize+padding) in BGR, and not simple-resize (no padding) RGB as in the second example.

Thanks again

**Comment by shaydeci** [comment]
If you want to use the same preprocessing that was used for validation- then yes padded rescale to 640X640 with BGR.
The dog was a square image anyway and the model handled it.
You can see the exact preprocessing that is used for validation for any recipe if you check their dataset_params, for instance these are the [dataset_params that were used for training YoloX](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/dataset_params/coco_detection_dataset_params.yaml), and
notice the [transforms](https://github.com/Deci-AI/super-gradients/blob/361e2ceb1b99af70658fdce11037d55774dca53d/src/super_gradients/recipes/dataset_params/coco_detection_dataset_params.yaml#L64) part inside valid_dataset_params.
